### PR TITLE
Initial support for Python & .NET

### PR DESCRIPTION
**Pull request by justinvp**
This adds initial support for Python and .NET with an [initial basic surface area](https://github.com/pulumi/pulumi-eks/issues/439#issuecomment-718273818).

Probably easiest to review commit by commit:

1. Move examples
2. Initial support for Python & .NET
3. Add examples
4. Split out tests for each language
5. Hookup Python and .NET tests
6. Use a simple golang program to launch node
7. Update CHANGELOG.md

Follow-up tasks (separate PRs):

- Add Go (Part of https://github.com/pulumi/pulumi-eks/issues/439; draft PR https://github.com/pulumi/pulumi-eks/pull/457, but blocked on https://github.com/pulumi/pulumi/pull/5497)
- Hook up GHA for publishing the plugin (https://github.com/pulumi/pulumi-eks/issues/456; WIP)
- Port dynamic provider resources to be implemented as custom resources of the new provider plugin (https://github.com/pulumi/pulumi-eks/issues/451; draft PR https://github.com/pulumi/pulumi-eks/pull/458)
- Schematize remaining components/properties (https://github.com/pulumi/pulumi-eks/issues/454; most of this is blocked on underlying support for externally referenced resources/types)
- Generate resource-based docs for EKS (https://github.com/pulumi/pulumi-eks/issues/447)
- Add more examples/tests (https://github.com/pulumi/pulumi-eks/issues/455)
- Expose functions / methods (https://github.com/pulumi/pulumi-eks/issues/452)

Part of https://github.com/pulumi/pulumi-eks/issues/439

### CODE REVIEW

**Comment by justinvp** [comment]
@pgavlin, @lblackstone this is ready to review

File: Makefile [hunk 1]
```diff
@@ -5,26 +5,71 @@ TESTPARALLELISM := 12
...
+		rm ./bin/setup.py.bak && \
+		cd ./bin && python3 setup.py build sdist
+
+build_dotnet:: DOTNET_VERSION := $(shell pulumictl get version --language dotnet)
+build_dotnet:: schema
+	rm -rf dotnet
+	cd provider/cmd/pulumi-gen-eks && go run main.go dotnet ../../../dotnet ../pulumi-resource-eks/schema.json $(VERSION)
+	cd dotnet/ && \
+		echo "${DOTNET_VERSION}" >version.txt && \
+        dotnet build /p:Version=${DOTNET_VERSION}
```
Comment (lblackstone) [inline Makefile:47]: This spacing is odd

File: dotnet/Cluster.cs [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+        /// The number of worker nodes that should be running in the cluster. Defaults to 2.
+        /// </summary>
+        [Input("desiredCapacity")]
+        public Input<int>? DesiredCapacity { get; set; }
+
+        [Input("enabledClusterLogTypes")]
+        private InputList<string>? _enabledClusterLogTypes;
+
+        /// <summary>
+        /// Enable EKS control plane logging. This sends logs to cloudwatch. Possible list of values are: ["api", "audit", "authenticator", "controllerManager", "scheduler"]. By default it is off.
```
Comment (lblackstone) [inline dotnet/Cluster.cs:99]: I assume we'll want to make this a relaxed enum at some point.

File: dotnet/Pulumi.Eks.csproj [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  </ItemGroup>
+
+  <ItemGroup>
+    <EmbeddedResource Include="version.txt" />
+    <Content Include="version.txt" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Pulumi" Version="2.*" />
+    <PackageReference Include="Pulumi.Aws" Version="3.*" />
```
Comment (lblackstone) [inline dotnet/Pulumi.Eks.csproj:39]: Doesn't this also require pulumi-kubernetes?

File: provider/cmd/pulumi-gen-eks/main.go [hunk 4]
```diff
@@ -0,0 +1,695 @@
...
+					},
+				},
+			},
+		},
+
+		Language: map[string]json.RawMessage{
+			"csharp": rawMessage(map[string]interface{}{
+				"packageReferences": map[string]string{
+					"Pulumi":     "2.*",
+					"Pulumi.Aws": "3.*",
```
Comment (lblackstone) [inline provider/cmd/pulumi-gen-eks/main.go:597]: Pulumi.Kubernetes?

File: provider/cmd/pulumi-resource-eks/schema.json [hunk 5]
```diff
@@ -0,0 +1,368 @@
+{
```
Comment (lblackstone) [inline provider/cmd/pulumi-resource-eks/schema.json:1]: Just to confirm: this isn't currently used, right?

File: python/README.md [hunk 6]
```diff
@@ -0,0 +1,185 @@
+[![Build Status](https://travis-ci.com/pulumi/pulumi-eks.svg?token=eHg7Zp5zdDDJfTjY8ejq&branch=master)](https://travis-ci.com/pulumi/pulumi-eks)
```
Comment (lblackstone) [inline python/README.md:1]: A lot of this readme is specific to Node

File: Makefile (hunk 1 above)
Comment (justinvp) [reply Makefile]: Fixed

File: dotnet/Cluster.cs (hunk 2 above)
Comment (justinvp) [reply dotnet/Cluster.cs:99]: Yep, there are a few places where it'd be ideal to use enums. I have some TODOs in the code that generates the schema around this and just filed https://github.com/pulumi/pulumi-eks/issues/470 to track.

File: dotnet/Pulumi.Eks.csproj (hunk 3 above)
Comment (justinvp) [reply dotnet/Pulumi.Eks.csproj:39]: Yep. Fixed.

File: provider/cmd/pulumi-gen-eks/main.go (hunk 4 above)
Comment (justinvp) [reply provider/cmd/pulumi-gen-eks/main.go]: Fixed.

File: provider/cmd/pulumi-resource-eks/schema.json (hunk 5 above)
Comment (justinvp) [reply provider/cmd/pulumi-resource-eks/schema.json:1]: This file is generated and we check it in like we do in other repos. It is actually used. In the `Makefile` there's a `schema` target that generates this file and the `build_python` and `build_dotnet` targets depend on the `schema` target and do actually read-in the generated `schema.json` in order to generate the code.

https://github.com/pulumi/pulumi-eks/blob/b69072bb65b74f5a6c138031c0af5767980bb4d8/Makefile#L30-L32

File: python/README.md (hunk 6 above)
Comment (justinvp) [reply python/README.md:1]: Agreed. The codegen just pulls in the root README.md. I opened https://github.com/pulumi/pulumi-eks/issues/471 to track updating it.

**Review by stack72 — APPROVED** [review]
So @justinvp and I just talked - this is largely structural changes for now - nothing in nodejs has changed - the new functionality hasn't been hooked up yet

**Comment by xdays** [comment]
@justinvp do you have any plan about when to publish python package(pulumi-eks) to pypi?  Thanks!

**Comment by stack72** [comment]
Hi @xdays 

We are working through this and hope to get this published soon - we can't give an official date at this moment but I can confidently tell you that it's high on our priority list

Paul

**Pull request merged**